OsmAnd-java/src/main/java/net/osmand/util/Algorithms.java

```diff
@@ -1363,6 +1363,27 @@ public static <T> List<T> removeAllFromList(Collection<T> original, Collection<T
 		return copy;
 	}
 
+	public static <T> void addAllIfNotContains(Collection<T> collection, Collection<T> elements) {
```

Comment:
Maybe we can move utils methods that work with collections to separate utils class? Algorithms already have 1.5k rows...

Reply:
Will be fixed in a separate task: https://github.com/osmandapp/OsmAnd-Issues/issues/2370